src/dashboard/user-interface/scores/scores-route-trait.php

```diff
@@ -0,0 +1,234 @@
+						],
+						'taxonomy' => [
+							'required'          => false,
+							'type'              => 'string',
+							'default'           => '',
+							'sanitize_callback' => 'sanitize_text_field',
+						],
+						'term' => [
+							'required'          => false,
+							'type'              => 'integer',
+							'default'           => null,
+							'sanitize_callback' => static function ( $param ) {
+								return \intval( $param );
+							},
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Gets the scores of a specific content type.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The success or failure response.
+	 */
+	public function get_scores( WP_REST_Request $request ) {
+		$content_type = $this->get_content_type( $request['contentType'] );
+		if ( $content_type === null ) {
+			return new WP_REST_Response(
+				[
+					'error' => 'Invalid content type.',
+				],
+				400
+			);
+		}
+
+		$taxonomy = $this->get_taxonomy( $request['taxonomy'], $content_type );
+		if ( $request['taxonomy'] !== '' && $taxonomy === null ) {
+			return new WP_REST_Response(
+				[
+					'error' => 'Invalid taxonomy.',
+				],
+				400
+			);
+		}
+
+		if ( ! $this->validate_term( $request['term'], $taxonomy ) ) {
+			return new WP_REST_Response(
+				[
+					'error' => 'Invalid term.',
+				],
+				400
+			);
+		}
+
+		return new WP_REST_Response(
+			$this->scores_repository->get_scores( $content_type, $taxonomy, $request['term'] ),
+			200
+		);
+	}
+
+	/**
+	 * Gets the content type object.
+	 *
+	 * @param string $content_type The content type.
+	 *
+	 * @return Content_Type|null The content type object.
+	 */
+	protected function get_content_type( string $content_type ): ?Content_Type {
+		$content_types = $this->content_types_collector->get_content_types();
+
+		if ( isset( $content_types[ $content_type ] ) && \is_a( $content_types[ $content_type ], Content_Type::class ) ) {
+			return $content_types[ $content_type ];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Gets the taxonomy object.
+	 *
+	 * @param string       $taxonomy     The taxonomy.
+	 * @param Content_Type $content_type The content type that the taxonomy is filtering.
+	 *
+	 * @return Taxonomy|null The taxonomy object.
+	 */
+	protected function get_taxonomy( string $taxonomy, Content_Type $content_type ): ?Taxonomy {
+		if ( $taxonomy === '' ) {
+			return null;
+		}
+
+		$valid_taxonomy = $this->taxonomies_repository->get_content_type_taxonomy( $content_type->get_name() );
+
+		if ( $valid_taxonomy && $valid_taxonomy->get_name() === $taxonomy ) {
+			return $valid_taxonomy;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Validates the term against the given taxonomy.
+	 *
+	 * @param int           $term_id  The ID of the term.
+	 * @param Taxonomy|null $taxonomy The taxonomy.
+	 *
+	 * @return bool Whether the term passed validation.
+	 */
+	protected function validate_term( ?int $term_id, ?Taxonomy $taxonomy ): bool {
```

Comment:
I am thinking $taxonomy is not allowed to be null here since before you call this its already checked right?

Reply:
That's not true. `$taxonomy` is allowed to be null when there's no taxonomy provided in the endpoint (aka, when `$request['taxonomy'] = 0`.